src/lib.rs

```diff
@@ -2777,6 +2777,161 @@ pub fn is_maximal_matching(
     })
 }
 
+fn _graph_triangles(graph: &graph::PyGraph, node: usize) -> (usize, usize) {
+    let mut triangles: usize = 0;
+
+    let index = NodeIndex::new(node);
+    let mut neighbors: HashSet<NodeIndex> =
+        graph.graph.neighbors(index).collect();
+    neighbors.remove(&index);
+
+    for nodev in &neighbors {
+        triangles += graph
+            .graph
+            .neighbors(*nodev)
+            .filter(|&x| (x != *nodev) && neighbors.contains(&x))
+            .count();
+    }
+
+    let d: usize = neighbors.len();
+    let triples: usize = match d {
+        0 => 0,
+        _ => (d * (d - 1)) / 2,
+    };
+
+    (triangles / 2, triples)
+}
+
+/// Compute the transitivity of an undirected graph.
+///
+/// The transitivity of a graph is defined as:
+///
+/// .. math::
+///     `c=3 \times \frac{\text{number of triangles}}{\text{number of connected triples}}`
+///
+/// A “connected triple” means a single vertex with
+/// edges running to an unordered pair of others.
```

Comment:
I think we also need to document something about the multithreaded nature of this function. I would add something here about the thread pool usage in rayon and how to adjust it. Basically just copy and adapt what I used here:

https://github.com/Qiskit/retworkx/pull/278/files#diff-b8d56121fa60f981d2a35720517aadf6cd87bd0ade23585819d1f84b3262eb54R343-R347